src/components/Tool/Appeal/InitialPage/AddAppealForm/AddAppealForm.tsx

```diff
@@ -0,0 +1,661 @@
+    tags: never[];
+    exclusions: ContactExclusion[];
+  }>
+>;
+
+const useStyles = makeStyles()((theme: Theme) => ({
+  loadingIndicator: {
+    margin: theme.spacing(0, 1, 0, 0),
+  },
+  input: {
+    width: '100%',
+  },
+  form: {
+    width: '100%',
+  },
+  submitButton: {
+    backgroundColor: theme.palette.mpdxBlue.main,
+    width: '150px',
+    color: 'white',
+  },
+  selectAll: {
+    color: theme.palette.mpdxBlue.main,
+    marginLeft: 5,
+    '&:hover': {
+      cursor: 'pointer',
+      textDecoration: 'underline',
+    },
+  },
+}));
+
+export interface AddAppealFormProps {
+  accountListId: string;
+  appealName?: string;
+  appealGoal?: number;
+  appealStatuses?: Pick<FilterOption, 'name' | 'value'>[];
+  appealExcludes?: ContactExclusion[];
+  appealIncludes?: object;
+  formRef?: React.MutableRefObject<FormikValues | undefined>;
+}
+
+const AddAppealForm: React.FC<AddAppealFormProps> = ({
+  accountListId,
+  appealName,
+  appealGoal,
+  appealStatuses,
+  appealExcludes,
+  appealIncludes = {},
+  formRef,
+}) => {
+  const { classes } = useStyles();
+  const { t } = useTranslation();
+  const { push } = useRouter();
+  const { enqueueSnackbar } = useSnackbar();
+  const { data: contactFilterTags, loading: loadingTags } = useContactTagsQuery(
+    {
+      variables: { accountListId },
+    },
+  );
+  const { data: contactFilterGroups, loading: loadingStatuses } =
+    useContactFiltersQuery({
+      variables: {
+        accountListId,
+      },
+      context: {
+        doNotBatch: true,
+      },
+    });
+  const [createNewAppeal, { loading: updating }] = useCreateAppealMutation();
+
+  const contactStatuses = useMemo(() => {
+    if (contactFilterGroups?.accountList?.contactFilterGroups) {
+      return (
+        contactFilterGroups.accountList.contactFilterGroups
+          .find((group) => group?.filters[0]?.filterKey === 'status')
+          ?.filters.find(
+            (filter: { filterKey: string }) => filter.filterKey === 'status',
+          ) as MultiselectFilter
+      ).options;
+    } else {
+      return [{ name: '', value: '' }];
+    }
+  }, [contactFilterGroups]);
+
+  const onSubmit = async (props: Attributes) => {
+    const attributes: AppealCreateInput = {
+      name: props.name,
+      amount: calculateGoal(
+        props.initialGoal,
+        props.letterCost,
+        props.adminCost,
+      ),
+    };
+
+    const inclusionFilter = buildInclusionFilter({
+      appealIncludes,
+      tags: props.tags,
+      statuses: props.statuses,
+    });
+    if (!isEqual(inclusionFilter, {})) {
+      attributes.inclusionFilter = inclusionFilter;
+    }
+    const exclusionFilter = buildExclusionFilter(props.exclusions);
+    if (!isEqual(exclusionFilter, {})) {
+      attributes.exclusionFilter = exclusionFilter;
+    }
+
+    await createNewAppeal({
+      variables: {
+        accountListId,
+        attributes,
+      },
+      update: (_, result) => {
```

Comment:
Since we're not updating the cache here, lets get the result from the promise that `createNewAppeal` and move this logic down after the call to `createNewAppeal`.

Reply:
I put it into the onComplete callback, is that okay?

Reply:
```
 await createNewAppeal({
      variables: {
        accountListId,
        attributes,
      },
      onCompleted: (data) => {
        push(
          `/accountLists/${accountListId}/tools/appeals/appeal/${data?.createAppeal?.appeal?.id}/list/tour`,
        );
      },
    });
    ```

Reply:
Yeah, that works too